src/particle/pdgid/functions.py

```diff
@@ -369,7 +375,7 @@ def is_dyon(pdgid):
         return False
     if _digit(pdgid, Location.Nr) != 1:
         return False
-    if _digit(pdgid, Location.Nl) != 1 and _digit(pdgid, Location.Nl) != 2:
+    if _digit(pdgid, Location.Nl) not in (1, 2):
```

Comment:
```suggestion
    if _digit(pdgid, Location.Nl) not in {1, 2}:
```

Set checking is faster (okay, maybe not even measurably faster when there are only two items, but on principle it's better). You could also use range.